src/queue.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file
+ * @brief     FIFO Queue.
+ *
+ * Queue without using the internal heap.
+ * Based on the following code: https://github.com/sdesalas/Arduino-Queue.h/blob/master/Queue.h
+ * @author    Wiebe van Breukelen, Steven de Salas
+ * @license   See LICENSE
+ */
+
+#ifndef QUEUE_HPP
+#define QUEUE_HPP
+
+#include "wrap-hwlib.hpp"
+
+template <class T, size_t QUEUE_SIZE>
+class Queue {
+  private:
+    unsigned int _front, _back, _count;
+    T _data[QUEUE_SIZE];
```

Comment:
Als het mogelijk is, probeer gebruik te maken van de C++ array.

Reply:
Good point. Is it OK for you that we will change this within the latest codebase ([the feature-uart-gripper](https://github.com/R2D2-2018/ROBOT_CLAW/tree/feature-uart-gripper) branch)?

Reply:
No, the point of these reviews is to make sure only code that is up to the right standards are in the master. So even if it's already changed in some other branch, it should also be part of the pull-request.  
If you already have new commits in the latest branch you can just make a new temporary branch for pull-request fixes.

Reply:
Ok, I just tested you the use of a C++ array. Unfortunately, in this function it cannot been (easly) done. As this method ```getUarmFirmwareVersion(char response[15])``` is used for printing the current firmware version located on the uArm, it uses ```hwlib::cout```. As hwlib::cout does not have an operator for a ```std::array``` or ```std::string``` this is not possible. However we could write it by ourselves, but in my opinion this would create unwanted overhead. In our case, an char[] style array would be the best fit.

Reply:
If there is a good reason for using a char[] array (like this), then it's fine.